magpy/gui/magpy_gui.py

```diff
@@ -1931,6 +1945,58 @@ def OnOpenURL(self, event):
             self.changeStatusbar("Loading url failed ... Ready")
             dlg.Destroy()
 
+    def OnOpenWebService(self, event):
+        stream = DataStream()
+        success = False
+        bookmarks = self.options.get('bookmarks',[])
+        ids = self.options.get('edgeid',[])
+        types = self.options.get('edgetype',[])
+        formats = self.options.get('edgeformat',[])
+        dlg = ConnectWebServiceDialog(None, title='Create and Open URL for Geomag Web Service', ids=ids, types=types, formats=formats)
+        if dlg.ShowModal() == wx.ID_OK:
+            url = dlg.url
+            start = dlg.starttime
+            end = dlg.endtime
+            elements = dlg.elements
+            sampling_period = dlg.sampling_period
+            if start < end:
+                self.changeStatusbar("Loading data ... be patient")
+                try:
+                    if "geomag.usgs.gov/ws/edge" in url:
```

Comment:
See comment below, ideally this could be done for web service requests in general; there will be non "geomag.usgs.gov" urls that would still benefit from this approach.